### DISCUSSION TITLE
What's the parameter of UMI-guided assembly

### DISCUSSION THREAD

**Discussion by Angieiscute** [milaboratory/mixcr#1680] [Q&A] [answered]
When we are using the following command, we notice the reads with the same UMI did not assemble into 1 read. How can we perform the UMI-guided assembly?
Thanks!

```
mixcr analyze generic-amplicon-with-umi \
--species hsa \
--tag-pattern "^(R1:*)\^N{1:19}(R2:*)" \
--rna \
--rigid-left-alignment-boundary \
--floating-right-alignment-boundary C \
input_R1.fastq.gz \
input_R2.fastq.gz \
result
```

**Comment by mizraelson** [comment]
Where does the UMI located in the read?

**Reply by Angieiscute** [reply]
on the 5 end of R2.

**Reply by mizraelson** [reply]
So it is the beginning of R2? how many nucleotides?

**Reply by Angieiscute** [reply]
the UMI is 14bp and it starts from the 20th bp on R2.

**Reply by mizraelson** [reply]
Then the command should be:

```
mixcr analyze generic-amplicon-with-umi \
--species hsa \
--tag-pattern "^(R1:*)\^N{19}(UMI:N{14})(R2:*)" \
--rna \
--rigid-left-alignment-boundary \
--floating-right-alignment-boundary C \
input_R1.fastq.gz \
input_R2.fastq.gz \
result
```

**Reply by Angieiscute** [reply]
But with this parameter, the reads with the same UMI will not be assembled into one read. We need to perform UMI-guided assembly which means, all reads are grouped by their UMI.

**Reply by mizraelson** [reply, marked as answer]
Yes, the command does exactly this. For each UMI group a consensus is created.

**Reply by Angieiscute** [reply]
We found a fact that maybe the consensus is created, but the count within the group is not corrected into 1 read.

**Reply by mizraelson** [reply]
Can you show an example? The number of reads stays the same, but you can refer to the number of UMIs. It does not become a single read as it is still important to know that some UMIs have more reads than the others. But the errors are corrected and you can use the UMI count (uniqueMoleculeCount) instead of the read count.

**Reply by Angieiscute** [reply]
Thanks for your help. uniqueMoleculeCount and uniqueMeculeFraction can be referred as the result after the Consensus assembled?

**Reply by mizraelson** [reply]
Yes. ’uniqueMoleculeCount’ is the number of unique UMI groups for the clone, which is the number of consensuses (collapsed reads).

**Reply by Angieiscute** [reply]
Thanks!
